packages/block-editor/src/components/inner-blocks/use-nested-settings-update.js

```diff
@@ -98,7 +101,29 @@ export default function useNestedSettingsUpdate(
 		}
 
 		if ( ! isShallowEqual( blockListSettings, newSettings ) ) {
-			updateBlockListSettings( clientId, newSettings );
+			// This is an optimization for the initial rendering of a block list
```

Comment:
This comment reads a bit oddly to me. A suggestion:

```suggestion
			// Batch updates to block list settings to avoid triggering cascading renders 
			// for each container block included in a tree and optimize initial render.
```